app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -409,9 +413,26 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         }
 
         if (!browserOverlay.isVisible && session.isYoutubeTV &&
-                event.keyCode == KeyEvent.KEYCODE_BACK) {
-            val escKeyEvent = KeyEvent(event.action, KeyEvent.KEYCODE_ESCAPE)
-            (activity as MainActivity).dispatchKeyEvent(escKeyEvent)
+                event.keyCode == KeyEvent.KEYCODE_BACK &&
+                event.action == KeyEvent.ACTION_DOWN) {
+            webView?.evalJS("""
+                (function () {
+                    return document.activeElement.parentElement.parentElement.id === 'guide-list';
+                })();
+                """,
+                    ValueCallback {
+                        val keyCode = if (it == "true") {
+                            val wv = (webView as ViewGroup).getChildAt(0) as WebView
+                            val backForward = wv.copyBackForwardList().toStringList()
+                            val youtubeIndex = backForward.lastIndexOf("https://ftv.cdn.mozilla.net/ytht")
+                            val goBackSteps = backForward.size - youtubeIndex
+                            wv.goBackOrForward(-goBackSteps)
+                            KeyEvent.KEYCODE_BACK
+                        } else KeyEvent.KEYCODE_ESCAPE
+
+                        val newKeyEvent = KeyEvent(KeyEvent.ACTION_UP, keyCode)
+                        (activity as MainActivity).dispatchKeyEvent(newKeyEvent)
```

Comment:
iiuc, I see a few problems with this:
- We're not dispatching the down event. That might work in this case but I'm concerned what side effects there might be
- We're dispatching the up event twice.
- We're not converting the BACK, UP physical key press into an ESC, UP physical key press

fwiw, I know we're pressed for time but I think it's think it's these types of edge cases that get uncovered while writing tests for code like this.

To be explicit, I think the logic we want here is:
```
if (youtube.com/tv) {
    if (!atYouTubeTopLevel) {
        dispatchHomeEvent(event.action)
    } else {
        goBackBeforeYouTube()
    }
} else {
    dispatchKeyEventAsNormal()
}
```